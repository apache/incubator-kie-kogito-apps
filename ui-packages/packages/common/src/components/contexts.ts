import React from 'react';

export const aboutLogoContext = React.createContext('');
