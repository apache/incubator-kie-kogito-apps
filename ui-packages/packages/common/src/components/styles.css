.kogito-common--aboutModalBox .pf-c-about-modal-box__hero {
    background-attachment: unset;
}

.kogito-common--PageLayout .pf-c-page__header-brand-link .pf-c-brand {
    --pf-c-page__header-brand-link--c-brand--MaxHeight: 42px;
}

.kogito-common__load-more {
    margin-bottom: var(--pf-global--spacer--lg);
    text-align: center;
}

.kogito-common--domain-explorer__table td:not(:first-of-type) {
    min-width: 10em;
}

.kogito-common__embedded-table  {
    --pf-c-table--nested--first-last-child--PaddingLeft: 0;
    margin-bottom: var(--pf-global--spacer--lg);
    max-width: 80vw;
}

.kogito-common--domain-explorer__table .pf-c-table__toggle{
    padding: 24px;
}

.kogito-common--domain-explorer__table-OverFlow {
    overflow-y: auto;
} 

.kogito-common--manage-columns__data-list .pf-c-data-list__toggle {
    margin-left: calc(-1 * var(--pf-global--spacer--md));
}

.kogito-common--manage-columns__data-list-content:not(:last-of-type) {
    box-shadow: none;   
}

.kogito-common--manage-columns__data-list .kogito-common--manage-columns__data-list-content .pf-c-data-list__expandable-content-body {
    padding: 0px;   
}

.kogito-common--manage-columns__data-list-content > .pf-c-data-list__expandable-content-body .pf-c-data-list__item-row {
    padding-left: calc( var(--pf-global--spacer--2xl) + var(--pf-global--spacer--lg));
}

.kogito-common--manage-columns__data-list-content .pf-c-data-list__cell {
    padding-left: var(--pf-global--spacer--md);
}

.kogito-common--manage-columns__modal .pf-c-dropdown {
    margin-bottom: var(--pf-global--spacer--md);
} 

.kogito-common--filter-options__inputs {
    margin-right: var(--pf-global--spacer--md);
}

.kogito-common--filter-options__inputs .pf-c-select__toggle-wrapper{
    min-width: 6em;
}