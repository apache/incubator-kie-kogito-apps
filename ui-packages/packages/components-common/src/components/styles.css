.kogito-components-common__load-more {
  margin-bottom: var(--pf-global--spacer--lg) !important;
  text-align: center;
}

.pf-c-spinner .pf-m-md .kogito-components-common__load-more-spinner {
  overflow: hidden;
  margin-left: 5px;
}

.kogito-components-common__form-footer-padding-top {
  padding-top: 20px;
}
