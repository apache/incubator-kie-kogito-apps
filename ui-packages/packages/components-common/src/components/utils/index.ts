export * from './FormActionsUtils';
export * from './FormValidator';
export * from './ModelConversionTool';
