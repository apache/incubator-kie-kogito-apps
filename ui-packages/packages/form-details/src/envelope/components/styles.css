.pf-c-code-editor__controls > button:nth-child(1) {
  color: #06c !important;
}

.kogito-form-details-container {
  height: 100%;
}
