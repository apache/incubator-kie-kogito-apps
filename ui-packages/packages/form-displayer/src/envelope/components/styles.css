.kogito-form-displayer__ball-beats {
  margin-top: 50% !important;
}

.kogito-form-displayer-error-wrapper__align-left {
  text-align: left !important;
}
