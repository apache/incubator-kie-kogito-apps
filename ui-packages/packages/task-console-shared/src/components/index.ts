export { default as TaskState } from './TaskState/TaskState';
