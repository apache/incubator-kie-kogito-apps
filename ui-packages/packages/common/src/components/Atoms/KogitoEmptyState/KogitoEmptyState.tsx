import React from 'react';
import {
  Bullseye,
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title
} from '@patternfly/react-core';
import {
  ExclamationTriangleIcon,
  InfoCircleIcon,
  SearchIcon
} from '@patternfly/react-icons';
import '@patternfly/patternfly/patternfly-addons.css';
import { OUIAProps, componentOuiaProps } from '../../../utils/OuiaUtils';

export enum KogitoEmptyStateType {
  Search,
  Refresh,
  Reset,
  Info
}

interface IOwnProps {
  type: KogitoEmptyStateType;
  title: string;
  body: string;
  onClick?: () => void;
}

export const KogitoEmptyState: React.FC<IOwnProps & OUIAProps> = ({
  type,
  title,
  body,
  onClick,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <Bullseye {...componentOuiaProps(ouiaId, 'kogito-empty-state', ouiaSafe)}>
      <EmptyState variant={EmptyStateVariant.full}>
        {type === KogitoEmptyStateType.Search && (
          <EmptyStateIcon icon={SearchIcon} size="sm" />
        )}
        {(type === KogitoEmptyStateType.Refresh ||
          type === KogitoEmptyStateType.Reset) && (
          <EmptyStateIcon
            icon={ExclamationTriangleIcon}
            size="sm"
            color="var(--pf-global--warning-color--100)"
          />
        )}
        {type === KogitoEmptyStateType.Info && (
          <EmptyStateIcon
            icon={InfoCircleIcon}
            size="sm"
            color="var(--pf-global--info-color--100)"
          />
        )}

        <Title headingLevel="h5" size="lg">
          {title}
        </Title>

        <EmptyStateBody>{body}</EmptyStateBody>

        {type === KogitoEmptyStateType.Refresh && (
          <Button variant="primary" onClick={onClick}>
            Refresh
          </Button>
        )}

        {type === KogitoEmptyStateType.Reset && (
          <Button variant="link" onClick={onClick}>
            Reset to default
          </Button>
        )}
      </EmptyState>
    </Bullseye>
  );
};
