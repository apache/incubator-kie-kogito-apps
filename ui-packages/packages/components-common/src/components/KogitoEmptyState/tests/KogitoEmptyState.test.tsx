/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from '@patternfly/react-core/dist/js/components/Button';
import { KogitoEmptyState, KogitoEmptyStateType } from '../KogitoEmptyState';

const MockedComponent = (): React.ReactElement => {
  return <></>;
};

jest.mock('@patternfly/react-icons/dist/js/icons/info-circle-icon', () =>
  Object.assign(
    { __esModule: true },
    jest.requireActual('@patternfly/react-icons'),
    {
      InfoCircleIcon: () => {
        return <MockedComponent />;
      }
    }
  )
);

jest.mock('@patternfly/react-icons/dist/js/icons/search-icon', () =>
  Object.assign(
    { __esModule: true },
    jest.requireActual('@patternfly/react-icons'),
    {
      SearchIcon: () => {
        return <MockedComponent />;
      }
    }
  )
);

jest.mock(
  '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon',
  () =>
    Object.assign(
      { __esModule: true },
      jest.requireActual('@patternfly/react-icons'),
      {
        ExclamationTriangleIcon: () => {
          return <MockedComponent />;
        }
      }
    )
);

const props = {
  title: 'No child process instances',
  body: 'This process has no related sub processes',
  ouiaId: 'empty-state-ouia-id'
};

describe('KogitoEmptyState component tests', () => {
  it('Search test', () => {
    const { container } = render(
      <KogitoEmptyState type={KogitoEmptyStateType.Search} {...props} />
    );
    expect(container).toMatchSnapshot();
  });
  it('Reset test', () => {
    const click = jest.fn();
    const { container } = render(
      <KogitoEmptyState
        type={KogitoEmptyStateType.Reset}
        onClick={click}
        {...props}
      />
    );
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByText('Reset to default'));
    expect(click).toHaveBeenCalledTimes(1);
  });
  it('Info test', () => {
    const { container } = render(
      <KogitoEmptyState type={KogitoEmptyStateType.Info} {...props} />
    );
    expect(container).toMatchSnapshot();
  });
  it('Refresh test', () => {
    const click = jest.fn();
    const { container } = render(
      <KogitoEmptyState
        type={KogitoEmptyStateType.Refresh}
        onClick={click}
        {...props}
      />
    );
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByText('Refresh'));
    expect(click).toHaveBeenCalledTimes(1);
  });
});
