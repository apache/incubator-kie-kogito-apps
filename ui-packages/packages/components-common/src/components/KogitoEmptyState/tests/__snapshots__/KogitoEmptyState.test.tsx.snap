// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KogitoEmptyState component tests Info test 1`] = `
<Component
  body="This process has no related sub processes"
  ouiaId="empty-state-ouia-id"
  title="No child process instances"
  type={3}
>
  <Bullseye
    data-ouia-component-id="empty-state-ouia-id"
    data-ouia-component-type="kogito-empty-state"
    data-ouia-safe={true}
  >
    <div
      className="pf-l-bullseye"
      data-ouia-component-id="empty-state-ouia-id"
      data-ouia-component-type="kogito-empty-state"
      data-ouia-safe={true}
    >
      <EmptyState
        variant="full"
      >
        <div
          className="pf-c-empty-state"
        >
          <div
            className="pf-c-empty-state__content"
          >
            <EmptyStateIcon
              color="var(--pf-global--info-color--100)"
              icon={[Function]}
            >
              <InfoCircleIcon
                aria-hidden="true"
                className="pf-c-empty-state__icon"
                color="var(--pf-global--info-color--100)"
              >
                <MockedComponent />
              </InfoCircleIcon>
            </EmptyStateIcon>
            <Title
              headingLevel="h5"
              size="lg"
            >
              <h5
                className="pf-c-title pf-m-lg"
              >
                No child process instances
              </h5>
            </Title>
            <EmptyStateBody>
              <div
                className="pf-c-empty-state__body"
              >
                This process has no related sub processes
              </div>
            </EmptyStateBody>
          </div>
        </div>
      </EmptyState>
    </div>
  </Bullseye>
</Component>
`;

exports[`KogitoEmptyState component tests Refresh test 1`] = `
<Component
  body="This process has no related sub processes"
  onClick={[MockFunction]}
  ouiaId="empty-state-ouia-id"
  title="No child process instances"
  type={1}
>
  <Bullseye
    data-ouia-component-id="empty-state-ouia-id"
    data-ouia-component-type="kogito-empty-state"
    data-ouia-safe={true}
  >
    <div
      className="pf-l-bullseye"
      data-ouia-component-id="empty-state-ouia-id"
      data-ouia-component-type="kogito-empty-state"
      data-ouia-safe={true}
    >
      <EmptyState
        variant="full"
      >
        <div
          className="pf-c-empty-state"
        >
          <div
            className="pf-c-empty-state__content"
          >
            <EmptyStateIcon
              color="var(--pf-global--warning-color--100)"
              icon={[Function]}
            >
              <ExclamationTriangleIcon
                aria-hidden="true"
                className="pf-c-empty-state__icon"
                color="var(--pf-global--warning-color--100)"
              >
                <MockedComponent />
              </ExclamationTriangleIcon>
            </EmptyStateIcon>
            <Title
              headingLevel="h5"
              size="lg"
            >
              <h5
                className="pf-c-title pf-m-lg"
              >
                No child process instances
              </h5>
            </Title>
            <EmptyStateBody>
              <div
                className="pf-c-empty-state__body"
              >
                This process has no related sub processes
              </div>
            </EmptyStateBody>
            <Button
              onClick={[MockFunction]}
              variant="primary"
            >
              <button
                aria-disabled={false}
                aria-label={null}
                className="pf-c-button pf-m-primary"
                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[MockFunction]}
                role={null}
                type="button"
              >
                Refresh
              </button>
            </Button>
          </div>
        </div>
      </EmptyState>
    </div>
  </Bullseye>
</Component>
`;

exports[`KogitoEmptyState component tests Reset test 1`] = `
<Component
  body="This process has no related sub processes"
  onClick={[MockFunction]}
  ouiaId="empty-state-ouia-id"
  title="No child process instances"
  type={2}
>
  <Bullseye
    data-ouia-component-id="empty-state-ouia-id"
    data-ouia-component-type="kogito-empty-state"
    data-ouia-safe={true}
  >
    <div
      className="pf-l-bullseye"
      data-ouia-component-id="empty-state-ouia-id"
      data-ouia-component-type="kogito-empty-state"
      data-ouia-safe={true}
    >
      <EmptyState
        variant="full"
      >
        <div
          className="pf-c-empty-state"
        >
          <div
            className="pf-c-empty-state__content"
          >
            <EmptyStateIcon
              color="var(--pf-global--warning-color--100)"
              icon={[Function]}
            >
              <ExclamationTriangleIcon
                aria-hidden="true"
                className="pf-c-empty-state__icon"
                color="var(--pf-global--warning-color--100)"
              >
                <MockedComponent />
              </ExclamationTriangleIcon>
            </EmptyStateIcon>
            <Title
              headingLevel="h5"
              size="lg"
            >
              <h5
                className="pf-c-title pf-m-lg"
              >
                No child process instances
              </h5>
            </Title>
            <EmptyStateBody>
              <div
                className="pf-c-empty-state__body"
              >
                This process has no related sub processes
              </div>
            </EmptyStateBody>
            <Button
              onClick={[MockFunction]}
              variant="link"
            >
              <button
                aria-disabled={false}
                aria-label={null}
                className="pf-c-button pf-m-link"
                data-ouia-component-id="OUIA-Generated-Button-link-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[MockFunction]}
                role={null}
                type="button"
              >
                Reset to default
              </button>
            </Button>
          </div>
        </div>
      </EmptyState>
    </div>
  </Bullseye>
</Component>
`;

exports[`KogitoEmptyState component tests Search test 1`] = `
<Component
  body="This process has no related sub processes"
  ouiaId="empty-state-ouia-id"
  title="No child process instances"
  type={0}
>
  <Bullseye
    data-ouia-component-id="empty-state-ouia-id"
    data-ouia-component-type="kogito-empty-state"
    data-ouia-safe={true}
  >
    <div
      className="pf-l-bullseye"
      data-ouia-component-id="empty-state-ouia-id"
      data-ouia-component-type="kogito-empty-state"
      data-ouia-safe={true}
    >
      <EmptyState
        variant="full"
      >
        <div
          className="pf-c-empty-state"
        >
          <div
            className="pf-c-empty-state__content"
          >
            <EmptyStateIcon
              icon={[Function]}
            >
              <SearchIcon
                aria-hidden="true"
                className="pf-c-empty-state__icon"
              >
                <MockedComponent />
              </SearchIcon>
            </EmptyStateIcon>
            <Title
              headingLevel="h5"
              size="lg"
            >
              <h5
                className="pf-c-title pf-m-lg"
              >
                No child process instances
              </h5>
            </Title>
            <EmptyStateBody>
              <div
                className="pf-c-empty-state__body"
              >
                This process has no related sub processes
              </div>
            </EmptyStateBody>
          </div>
        </div>
      </EmptyState>
    </div>
  </Bullseye>
</Component>
`;
